server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -1045,6 +1056,31 @@ public void testIntWithMinMax() {
         assertEquals(1, integerSetting.get(Settings.EMPTY).intValue());
     }
 
+    public void testIntegerParser() throws Exception {
+        String expectedKey = "test key";
+        int expectedMinValue = Integer.MIN_VALUE;
+        int expectedMaxValue = Integer.MAX_VALUE;
+        boolean expectedFilteredStatus = true;
+        IntegerParser integerParser = new IntegerParser(expectedMaxValue, expectedMinValue, expectedKey, expectedFilteredStatus);
```

Comment:
You swapped min and max here (and all the others), which is causing the tests to fail.